crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -49,8 +49,20 @@ pub struct Mesh {
 /// # use bevy_render::render_resource::PrimitiveTopology;
 /// fn create_triangle() -> Mesh {
 ///     let mut mesh = Mesh::new(PrimitiveTopology::TriangleList);
-///     mesh.insert_attribute(Mesh::ATTRIBUTE_POSITION, vec![[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [1.0, 1.0, 0.0]]);
-///     mesh.set_indices(Some(Indices::U32(vec![0,1,2])));
+///     mesh.insert_attribute(
+///         Mesh::ATTRIBUTE_POSITION,
+///         vec![[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [1.0, 1.0, 0.0], [0.0, 0.0, 0.0]]
+///     );
+///     // We have created 4 vertices, each with it's own position attribute.
+///     mesh.insert_attribute(
+///         Mesh::ATTRIBUTE_UV_0,
+///         vec![[0.0, 0.0], [1.0, 0.0], [0.0, 1.0], [1.0, 1.0]]
+///     );
```

Comment:
I think your UVs have a problem:
![image](https://github.com/bevyengine/bevy/assets/134181069/f37cee40-a567-463e-9e45-b1e156eead37)

Reply:
Also maybe adding a comment explaining that for `StandardMaterial` to work properly with a mesh, it should have:
- an `ATTRIBUTE_UV_0` (to display a texture (without it the mesh appears grey on my computer))
- an `ATTRIBUTE_NORMAL` (to be able to process light properly (without it the light seems to be ignored)), you could maybe add one in your example?